src/domain/models/Credential.ts

```diff
@@ -16,6 +16,12 @@ export abstract class Credential implements Pluto.Storable {
 
   public readonly uuid = Pluto.makeUUID();
 
+  isCredentialType<
+    C extends Credential
+  >(t: CredentialType): this is C {
+    return this.credentialType === t
+  }
```

Comment:
I don't think we should keep this, it's effectively a false cast that would be better served with `instanceOf`

Reply:
Solved